src/libs/typesystem/spectranslator/Elektra/Parsers.hs

```diff
@@ -0,0 +1,123 @@
+--
+-- @file
+--
+-- @brief Some parsers which are used in the translation process
+--
+-- To parse the type signatures which are used to carry all the
+-- required information to create a corresponding specification
+-- in our haskell EDSL we use a parser. As those type signatures
+-- are not necessarily regular we use a parser instead of regexes
+-- or string splitting to have a robust extensible foundation.
+--
+-- @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+--
+module Elektra.Parsers (parseTypeSignature, parseRange, 
+  regexTypeP, regexTypeParameterP, typeSignatureP, rangeP) where
+
+import Control.Applicative (empty)
+import Data.Void
+
+import Elektra.Specifications
+
+import Text.Megaparsec
+import Text.Megaparsec.Char
+import qualified Text.Megaparsec.Char.Lexer as L
+
+type Parser = Parsec Void String
+
+-- Parse the given strings without further error analysis for now
+
+parseTypeSignature :: String -> Maybe TypeSignature
+parseTypeSignature = parseMaybe typeSignatureP
+
+parseRange :: String -> Maybe (Int, Int)
+parseRange = parseMaybe rangeP
+
+-- Lexer for the signatures
```

Comment:
type signatures. Can you give an informal description how they look like? (With examples. Maybe in the README)

Reply:
added. they are mainly relevant for creators of checker plugins and similar things to describe their regex-based semantics. Normal users shouldn't have to deal with that.